google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -572,6 +574,12 @@ public void write(Iterable<LogEntry> logEntries, WriteOption... options) {
   public void flush() {
     // BUG(1795): We should force batcher to issue RPC call for buffered messages,
     // so the code below doesn't wait uselessly.
+    boolean flushBuffer = (pendingWrites.size() < BUFFER_SIZE && !closed) ? true : false;
+    if (flushBuffer) {
+      return;
+    } else if (FLUSH_ON_ERROR) {
+      return;
+    }
```

Comment:
The condition can be rewritten as below but you are not doing anything for either of the case except for returning then why another if check?.
```java
    if (pendingWrites.size() < BUFFER_SIZE && !closed) {
     // What would you do here?
      return;
    } else if (FLUSH_ON_ERROR) {
    // Here as well?
      return;
    }
```